frontend/src/components/maps/leaflet/Map.tsx

```diff
@@ -169,6 +159,19 @@ const Map: React.FC<MapProps> = ({
     setActiveBasemap(current);
   };
 
+  const onSingleMarkerClick = (point: PointFeature) => {
+    onMarkerClick?.(asProperty(point));
+  };
+
+  const onClusterClick = (cluster: ICluster, expansionZoom: number) => {
+    // zoom to cluster
+    const [longitude, latitude] = cluster?.geometry?.coordinates;
+    const leafletMap = mapRef?.current?.leafletElement;
+    leafletMap?.setView([latitude, longitude], expansionZoom, {
+      animate: true,
+    });
+  };
+
```

Comment:
Clicking on a map pin will show the popup like before. Clicking on a cluster will zoom the map to the extent of the cluster. 

While testing I noticed that a few cluster wouldn't "expand into single markers because all the buildings are on the SAME lat long location. I will address that in another PR